### PR TITLE
Fix checking maximum header size, do not take start of body into account

### PR DESCRIPTION
**Pull request by nopolabs**
Fixed check for exceeding maximum header size in RequestHeaderParser::feed()

### CODE REVIEW

**Comment by clue** [comment]
Thanks for filing this PR :+1: 

Unfortunately it has no tests, see also #82.

File: src/Server.php [hunk 1]
```diff
@@ -41,6 +41,9 @@ public function __construct(SocketServerInterface $io)
                     $conn->emit('resume');
                 });
             });
+            $parser->on('error', function($exception) {
+                $this->emit('error', [$exception]);
+            });
```
Comment (WyriHaximus) [inline src/Server.php]: Not sure why this is in this PR? (It does make sense to do this non the less.)

Comment (WyriHaximus) [inline src/Server.php]: FYI this has been addressed in #83

**Comment by nopolabs** [comment]
added test

**Comment by WyriHaximus** [comment]
> added test

Test look good, but now [this test](https://github.com/reactphp/http/blob/master/tests/ServerTest.php#L70-L90) is failing

> removed bubble up of error from Server.php, not related to this issue

Great, thanks 👍

File: tests/ServerTest.php [hunk 2]
```diff
@@ -81,8 +81,9 @@ public function testParserErrorEmitted()
         $conn = new ConnectionStub();
         $io->emit('connection', [$conn]);
 
-        $data = $this->createGetRequest();
-        $data = str_pad($data, 4096 * 4);
+        $data = chop($this->createGetRequest());
```
Comment (clue) [inline tests/ServerTest.php]: This looks *okay*, but could probably be a bit *clearer*. What do you think, should this be updated to use a literal request string here instead?

Comment (nopolabs) [inline tests/ServerTest.php]: I agree, thanks for the suggestion.

**Pull request merged**